deployment/postgres-init-db/sql/views/merlin/derived_events.sql

```diff
@@ -0,0 +1,106 @@
+                              from derivation_tb_range tr1
+                                  left join derivation_tb_range tr2 on tr1.valid_at < tr2.valid_at
+                              group by tr1.key, tr1.derivation_group_name, tr1.valid_at, tr1.dr
+                        )
+                        -- this final selection (sub) utilizes the first, as well as merlin.subtract_later_ranges, to produce a sparse multirange that a given source is valid over. See merlin.subtract_later_ranges for further details on subtracted ranges.
+                        select ranges_with_subs.key,
+                              ranges_with_subs.derivation_group_name,
+                              ranges_with_subs.original_range,
+                              ranges_with_subs.subsequent_ranges,
+                              merlin.subtract_later_ranges(ranges_with_subs.original_range, ranges_with_subs.subsequent_ranges) AS source_range,
+                              ranges_with_subs.valid_at
+                        from ranges_with_subs
+                        order by ranges_with_subs.derivation_group_name desc, ranges_with_subs.valid_at) sub on sub.key = external_event.source_key and sub.derivation_group_name = external_event.derivation_group_name
+                where sub.source_range @> external_event.start_time
+        order by sub.derivation_group_name, external_event.start_time) rule1_3) t
+where rn = 1
+order by start_time;
+
+-- create a unique index, which allows concurrent refreshes
+create unique index on merlin.derived_events (
+  event_key,
+  source_key,
+  derivation_group_name,
+  event_type_name
+);
+
+-- refresh the materialized view after insertion/update/deletion to external_source and external_event and derivation_group
+create function merlin.refresh_derived_events_on_trigger()
+  returns trigger
+  language plpgsql as $$
+begin
+  refresh materialized view concurrently merlin.derived_events;
+  return new;
+end;
+$$;
+
+-- events are the most basic source of information, so we have to trigger on their insertion
+create trigger refresh_derived_events_on_external_event
+after insert or update or delete on merlin.external_event
+  for each statement execute function merlin.refresh_derived_events_on_trigger();
+
+-- also trigger on external sources, especially in the case of an empty source, which could still overlap and erase some
+--    events in time (see "rule3_empty" test in ExternalEventTests.java).
+create trigger refresh_derived_events_on_external_source
+after insert or update or delete on merlin.external_source
+  for each statement execute function merlin.refresh_derived_events_on_trigger();
+
+create trigger refresh_derived_events_on_derivation_group
+after insert or update or delete on merlin.derivation_group
+  for each statement execute function merlin.refresh_derived_events_on_trigger();
+
+comment on materialized view merlin.derived_events is e''
+  'Details all derived events from all derivation groups.';
```

Comment:
Minor: wording suggestion:
```suggestion
comment on materialized view merlin.derived_events is e''
  'Derives the final event set for each derivation group';
```

It can be tweaked beyond this, but I think that wording may be a bit clearer.

Reply:
Fixed!